Contents/Code/plex_api_helper.py

```diff
@@ -92,3 +113,94 @@ def add_themes(rating_key, theme_files=None, theme_urls=None):
                     plex_item.uploadTheme(url=theme_url)
     else:
         Log.Info('No theme songs provided for rating key: %s' % rating_key)
+
+
+def plex_listener():
+    # type: () -> None
+    """
+    Listen for events from Plex server.
+
+    Send events to ``plex_listener_handler`` and errors to ``Log.Error``.
+
+    Examples
+    --------
+    >>> plex_listener()
+    ...
+    """
+    global plex
+    if not plex:
+        plex = setup_plexapi()
+    listener = plexapi.alert.AlertListener(server=plex, callback=plex_listener_handler, callbackError=Log.Error)
+    listener.run()
+
+
+def plex_listener_handler(data):
+    # type: (dict) -> None
+    """
+    Process events from ``plex_listener()``.
+
+    Check if ``data`` is for an event that indicates metadata update has finished for a movie item, and process further
+    as required.
+
+    Parameters
+    ----------
+    data : dict
+        Data received from the Plex server.
+
+    Examples
+    --------
+    >>> plex_listener_handler(data={'type': 'timeline'})
+    ...
+    """
+    global plex, processing
+    if not plex:
+        plex = setup_plexapi()
+    # Log.Debug(data)
+    if data['type'] == 'timeline':
+        for entry in data['TimelineEntry']:
+            if entry['state'] == 5 and entry['type'] == 1 and entry['identifier'] == 'com.plexapp.plugins.library':
```

Comment:
Use `reverseSearchType(entry['type']) == 'movie'` to make this more readable.

Add comment link for `https://github.com/pkkid/python-plexapi/blob/8b3235445f6b3051c39ff6d6fc5d49f4e674d576/plexapi/utils.py#L35-L55`

Add link to known state values: `https://python-plexapi.readthedocs.io/en/latest/modules/alert.html#module-plexapi.alert`